sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/internal/state/AsynchronousMetricStorage.java

```diff
@@ -168,6 +169,17 @@ public MetricData collectAndReset(
     }
   }
 
+  // Taken from RxJava throwIfFatal, which was taken from scala
+  private static void propagateIfFatal(Throwable t) {
```

Comment:
Yes, Scala `NonFatal` would come in handy here.
Nit: worth making it a public utility method for very possible further reuse.

Reply:
I don't think OpenTelemetry should be in the business of providing general utilities. :)

Reply:
I can definitely see this being used elsewhere so we can have an .internal helper. But can be later too

Reply:
By the way here's just one of many locations we would want to apply the new pattern

https://github.com/open-telemetry/opentelemetry-java/blob/main/sdk/trace/src/main/java/io/opentelemetry/sdk/trace/export/MultiSpanExporter.java#L45